File: src/main/java/yunas/YunasBatch.java
package yunas;

import yunas.batch.BatchManager;
import yunas.batch.DefaultBatchManager;
import yunas.configuration.Configuration;

/**
 * Entry Point of Yunas Framework For DefaultBatchManager.
 *
 */
public class YunasBatch  {

    private static final BatchManager manager = new DefaultBatchManager();

    private static final App instance = App.initBatch();

    public static Configuration getConfiguration() {
        return instance.getConfiguration();
    }

    public static void  run(String[] args) {
        manager.run(args);
    }

    public static void add(String name, Batch executable) {
        manager.add(name,executable);
    }

}

File: src/main/java/yunas/YunasBatch.kt
package yunas

import yunas.batch.DefaultBatchManager
import yunas.configuration.Configuration

/**
 * Entry Point of Yunas Framework For DefaultBatchManager.
 *
 */
object YunasBatch {

    private val manager = DefaultBatchManager()

    private val instance = App.initBatch()

    val configuration: Configuration?
        get() = instance.configuration

    @JvmStatic
    fun run(args: Array<String>) {
        manager.run(args)
    }

    fun add(name: String, executable: Batch) {
        manager.add(name, executable)
    }

}